### DISCUSSION TITLE
https://www.spsanderson.com/steveondata/posts/2024-10-29/

### DISCUSSION THREAD

**Discussion by giscus** [spsanderson/steveondata#15] [Comments]
# https://www.spsanderson.com/steveondata/posts/2024-10-29/

Learn how to create and manipulate lists in R with comprehensive examples. Perfect for beginners, covering basic to advanced list operations with practical code samples.

https://www.spsanderson.com/steveondata/posts/2024-10-29/

**Comment by spsanderson** [comment]
I was emailed today by Dr. Biggerstaff thankfully, in order to include another example of list creation. Namely this is using something like `vector(mode = "list", length = n)`

So without further adieu lets get to understand how to use vector() with mode="list" in R with the below example function. 

#### Basic Explanation
In R, `vector(mode = "list")` creates an empty list vector, while `vector(mode = "list", length = n)` creates a list vector of specified length n with NULL elements. Lists in R are versatile data structures that can hold elements of different types

Here's an example function that demonstrates the usage:

```R
create_student_records <- function(n) {
  # Initialize an empty list vector of length n
  students <- vector(mode = "list", length = n)
  
  # Populate each list element with a student record
  for(i in 1:n) {
    students[[i]] <- list(
      id = paste0("STU", i),
      name = NA,
      grades = numeric(0),
      active = TRUE
    )
  }
  
  return(students)
}

# Usage example:
student_records <- create_student_records(3)
student_records
[[1]]
[[1]]$id
[1] "STU1"

[[1]]$name
[1] NA

[[1]]$grades
numeric(0)

[[1]]$active
[1] TRUE


[[2]]
[[2]]$id
[1] "STU2"

[[2]]$name
[1] NA

[[2]]$grades
numeric(0)

[[2]]$active
[1] TRUE


[[3]]
[[3]]$id
[1] "STU3"

[[3]]$name
[1] NA

[[3]]$grades
numeric(0)

[[3]]$active
[1] TRUE

student_records[[1]][["id"]]
[1] "STU1"
```

#### How It Works
- The function creates a list vector of length `n` using `vector(mode = "list", length = n)`
- Each element in the list can contain different types of data (strings, numbers, vectors, logical values)
- Lists are more flexible than regular vectors because they can store elements of different types
- You can access and modify individual elements using double square brackets `[[]]` example function creates a structured list where each element represents a student record with multiple fields. This demonstrates how list vectors can be used to create complex data structures in R.
